mautic/mautic
# [Enhancement] Support includes/excludes with text fields for bulk filtering.

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #5924
| BC breaks? | No
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
See #5924

#### Steps to test this PR:
1. Create a new segment "Test includes/excludes segment"
2. Filter this segment by "Zip Code" with "includes" and put in several zip codes that you know some of your contacts have as a string. Any listing format should suffice. Example strings below.
3. Run `php console mautic:segments:update` or wait for your cron to execute.
4. The new segment should have the contacts that are within those zipcodes.

This should work with any text field, I'm just using ZipCodes as a real-world example.

Example: Random zips without spaces from the demo data. As copied by Numbers. Should auto-delimit by space.
```
27401 2870 30458 3263 3312 3342 33762 347 4357 451
```
Example: Some UK zips as produced by Excel when copying. Should auto-delimit by space using quote as the enclosure.
```
"GL15 1XD" "HS6 2YL" "IV28 9UH" "J0K 1X0" "J0R 1C0" "J4H 1M3" "K1H 7Z1" "L8G 2V1" "LA6 6ZT" "N9A 1H9" "PA31 8WG" "SY10 6ND" "SY22 2AD" "TA11 7UU" "TA18 7XS" "V0E 2P0" "V0K 1A0" "V6B 3K9" "ZE2 4FH"
```
Example: All zipcodes in demo data, copied from a CSV.
```
 "07020",10965,11201,1220,1512,1544,17036,2102,220,2337,27401,2870,30458,3263,3312,3342,33762,347,4357,451,4671,4860,49077,49221,50313,505,5185,57401,6204,6312,6475,705,7435,87112,"CH5 9LB","GL15 1XD","HS6 2YL","IV28 9UH","J0K 1X0","J0R 1C0","J4H 1M3","K1H 7Z1","L8G 2V1","LA6 6ZT","N9A 1H9","PA31 8WG","SY10 6ND","SY22 2AD","TA11 7UU","TA18 7XS","V0E 2P0","V0K 1A0","V6B 3K9","ZE2 4FH"
```
